tests/storage/test_extraction.py

```diff
@@ -201,7 +261,104 @@ def test_parameters(client):
         df = pd.read_csv(stream, index_col=0, float_precision="round_trip")
 
         # ERT produces a low-quality version
-        assert_almost_equal(parameters_df[col].values, df.values.flatten(), decimal=4)
+        assert_almost_equal(parameters_df[col].values, df.values.flatten(), decimal=5)
+
+
+def test_observations(client):
+    data = pd.DataFrame([[1, 0.1], [2, 0.2], [3, 0.3], [4, 0.4]], index=[2, 4, 6, 8])
+
+    builder = ErtConfigBuilder()
+    builder.add_general_observation("OBS", "RES", data)
+    ert = builder.build()
+
+    # Post ensemble
+    extraction.post_ensemble_data(ert, builder.ensemble_size)
+
+    # Experiment should have 1 observation
+    experiment_id = client.fetch_experiment()
+    observations = client.get(f"/experiments/{experiment_id}/observations").json()
+    assert len(observations) == 1
+
+    # Validate data
+    obs = observations[0]
+    assert obs["name"] == "OBS"
+    assert obs["values"] == data[0].tolist()
+    assert obs["errors"] == data[1].tolist()
+    assert obs["x_axis"] == data.index.astype(str).tolist()
+    assert obs["transformation"] is None
+
+
+def test_post_ensemble_results(client):
+    data = pd.DataFrame([[1, 0.1], [2, 0.2], [3, 0.3], [4, 0.4]], index=[2, 4, 6, 8])
+
+    # Add priors to ERT config
+    builder = ErtConfigBuilder()
+    builder.ensemble_size = 2
+    response_name = "RES"
```

Comment:
Style: Move `response_name` to before the `builder` maybe? I personally think it's nicer if we don't mix assigning to constants with other code.

Feel free to ignore. 👍